tests/shared/core/test_domain.py

```diff
@@ -1142,3 +1145,63 @@ def test_valid_slot_mappings(domain_as_dict: Dict[Text, Any]):
 def test_form_invalid_mappings(domain_as_dict: Dict[Text, Any]):
     with pytest.raises(InvalidDomain):
         Domain.from_dict(domain_as_dict)
+
+
+def test_slot_order_is_preserved():
```

Comment:
Can we please also add a test where we merge domains and then make sure that the slot order still makes sense? Merging is usually the part where things go wrong in this module 🙈